api/src/modules/h3-data/h3-data.service.ts

```diff
@@ -410,30 +408,46 @@ export class H3DataService {
       getImpactMapDto.supplierIds,
     );
 
+    const materialsH3DataYears: MaterialsH3DataYears[] = [];
+
+    if (getImpactMapDto.materialIds?.length)
+      for (const materialId of getImpactMapDto.materialIds) {
+        const material: Material = await this.materialService.getMaterialById(
```

Comment:
This makes roundtrips to the DB as many times as materials are requested from the consumer being a hit to performance. Given the data-intensive nature of this product, I think we should try to avoid as many DB calls as possible. Do you think we can improve this?